packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -91,7 +91,11 @@ export async function handleHotUpdate(
 	// Bugfix: sometimes style URLs get normalized and end with `lang.css=`
 	// These will cause full reloads, so filter them out here
 	const mods = [...filtered].filter((m) => !m.url.endsWith('='));
-	const file = ctx.file.replace(config.root.pathname, '/');
+	let file = ctx.file;
+	if (!file.startsWith('/')) {
+		file = `/${file}`
+	}
+	file = file.replace(config.root.pathname, '/')
```

Comment:
Nice! That fixes it.

It looks like it still shows the full path when the project name has a space, like in the case of the issue screenshot. Can you check what happens there?

Reply:
I handled only spaces, I don't know if there are any other special character that causes the issue